src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/MvcJsonMvcOptionsSetup.cs

```diff
@@ -29,38 +35,40 @@ public class MvcJsonMvcOptionsSetup : ConfigureOptions<MvcOptions>
             IOptions<MvcJsonOptions> jsonOptions,
             ArrayPool<char> charPool,
             ObjectPoolProvider objectPoolProvider)
-            : base((options) => ConfigureMvc(
-                options,
-                jsonOptions.Value.SerializerSettings,
-                loggerFactory,
-                charPool,
-                objectPoolProvider))
         {
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            if (jsonOptions == null)
+            {
+                throw new ArgumentNullException(nameof(jsonOptions));
+            }
+
+            _loggerFactory = loggerFactory;
```

Comment:
We should consistently null check all the things. This makes it seem like `charPool` and `objectPoolProvider` might be optional.